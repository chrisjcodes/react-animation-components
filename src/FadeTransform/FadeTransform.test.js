import React from 'react';
import FadeTranform from './index.jsx';
import renderer from 'react-test-renderer';

describe('FadeTranform', () => {
    test('renders', () => {
        const component = renderer.create(<FadeTranform />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('can accept custom styles', () => {
        const component = renderer.create(
            <FadeTranform style={{ background: 'black' }} />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionDelay with delay prop', () => {
        const component = renderer.create(<FadeTranform delay="1s" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionDuration with duration prop', () => {
        const component = renderer.create(<FadeTranform duration="1s" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets transitionTimingFunction with timingFn prop', () => {
        const component = renderer.create(
            <FadeTranform timingFn="ease-in-out" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets start', () => {
        const component = renderer.create(
            <FadeTranform start="translateX(10em)" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('sets finish', () => {
        const component = renderer.create(
            <FadeTranform finish="translateX(10em)" />
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
