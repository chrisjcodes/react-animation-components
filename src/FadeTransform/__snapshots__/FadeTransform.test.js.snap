// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FadeTranform can accept custom styles 1`] = `
<div
  style={
    Object {
      "background": "black",
      "opacity": 0,
      "transitionDelay": null,
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "linear",
    }
  }
>
  <div
    style={
      Object {
        "background": "black",
        "transform": "none",
        "transitionDelay": null,
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "linear",
      }
    }
  />
</div>
`;

exports[`FadeTranform renders 1`] = `
<div
  style={
    Object {
      "opacity": 0,
      "transitionDelay": null,
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "linear",
    }
  }
>
  <div
    style={
      Object {
        "transform": "none",
        "transitionDelay": null,
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "linear",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets finish 1`] = `
<div
  style={
    Object {
      "opacity": 0,
      "transitionDelay": null,
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "linear",
    }
  }
>
  <div
    style={
      Object {
        "transform": "none",
        "transitionDelay": null,
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "linear",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets start 1`] = `
<div
  style={
    Object {
      "opacity": 0,
      "transitionDelay": null,
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "linear",
    }
  }
>
  <div
    style={
      Object {
        "transform": "translateX(10em)",
        "transitionDelay": null,
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "linear",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets transitionDelay with delay prop 1`] = `
<div
  style={
    Object {
      "opacity": 0,
      "transitionDelay": "1s",
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "linear",
    }
  }
>
  <div
    style={
      Object {
        "transform": "none",
        "transitionDelay": "1s",
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "linear",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets transitionDuration with duration prop 1`] = `
<div
  style={
    Object {
      "opacity": 0,
      "transitionDelay": null,
      "transitionDuration": "1s",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "linear",
    }
  }
>
  <div
    style={
      Object {
        "transform": "none",
        "transitionDelay": null,
        "transitionDuration": "1s",
        "transitionProperty": "transform",
        "transitionTimingFunction": "linear",
      }
    }
  />
</div>
`;

exports[`FadeTranform sets transitionTimingFunction with timingFn prop 1`] = `
<div
  style={
    Object {
      "opacity": 0,
      "transitionDelay": null,
      "transitionDuration": "500ms",
      "transitionProperty": "opacity",
      "transitionTimingFunction": "ease-in-out",
    }
  }
>
  <div
    style={
      Object {
        "transform": "none",
        "transitionDelay": null,
        "transitionDuration": "500ms",
        "transitionProperty": "transform",
        "transitionTimingFunction": "ease-in-out",
      }
    }
  />
</div>
`;
